plugins/maven/src/test/java/org/jetbrains/idea/maven/execution/MavenJUnitPatcherTest.kt

```diff
@@ -36,6 +36,82 @@ class MavenJUnitPatcherTest : MavenMultiVersionImportingTestCase() {
     MavenProjectSettings.getInstance(myProject).testRunningSettings.isPassSystemProperties = true
   }
 
+  @Test
+  @Throws(CantRunException::class)
+  fun ExcludeProjectDependencyInClassPathElement() {
+    val m = createModulePom("m", """
+      <groupId>test</groupId>
+      <artifactId>m</artifactId>
+      <version>1</version>
+      <dependencies>
+        <dependency>
+          <groupId>test</groupId>
+          <artifactId>dep</artifactId>
+          <version>1</version>
+        </dependency>
+      </dependencies>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <version>2.16</version>
+            <configuration>
+              <classpathDependencyExcludes>
+                <classpathDependencyExclude>test:dep</classpathDependencyExclude>
+              </classpathDependencyExcludes>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+      """.trimIndent())
+
+    val dep = createModulePom("dep", """
+      <groupId>test</groupId>
+      <artifactId>dep</artifactId>
+      <version>1</version>
+      <dependencies>
+        <dependency>
+          <groupId>junit</groupId>
+          <artifactId>junit</artifactId>
+          <version>4.0</version>
+        </dependency>
+      </dependencies>
+      
+      """.trimIndent())
+
+    createProjectSubDirs("m/src/main/java",
+                         "m/target/classes",
+                         "dep/src/main/java",
+                         "dep/target/classes")
+
+    importProjects(m, dep)
+    assertModules("m", "dep")
+    assertModuleModuleDeps("m", "dep")
+
+    val module = getModule("m")
+
+    val mavenJUnitPatcher = MavenJUnitPatcher()
+    val javaParameters = JavaParameters()
+
+    val pathTransformer = { path: String ->
+      if (path.startsWith(projectPath)) {
+        path.replace(projectPath, "")
+      }
+      else {
+        path.substringAfterLast('/')
+      }
+    }
+
+    javaParameters.configureByModule(module, JavaParameters.CLASSES_AND_TESTS, IdeaTestUtil.getMockJdk18())
+    assertEquals(listOf("/m/target/classes", "/dep/target/classes", "junit-4.0.jar"),
```

Comment:
The test fails for me here because `classPath` is the full path and you assert it to the relative path here

Reply:
fixed

Reply:
It doesn't look fixed for me on the first assert:
```
Expected :[dep/target/classes, junit-4.0.jar, m/target/classes]
Actual   :[dep\target\classes, junit-4.0.jar, m\target\classes]
```

But even if I fix the forward/backslash issue here on second assert the result is:
```
Expected :[junit-4.0.jar, m\target\classes]
Actual   :[dep\target\classes, junit-4.0.jar, m\target\classes]
```

Reply:
Currently, I am unable to reproduce the second issue on my end. Could you provide some additional information, such as the environment and how you are executing the test case?

Reply:
Looks like it's ok on Linux/MacOS but not on Windows. I have found the issue, will merge your change and the issue with local paths myself.